AlphaWallet/UI/Views/AmountTextField.swift

```diff
@@ -454,14 +502,62 @@ extension Double {
     }
 }
 
-fileprivate extension String {
-    var withDecimalSeparatorReplacedByPeriod: String {
-        guard let decimalSeparator = Locale.current.decimalSeparator else { return self }
-        let period = "."
-        if decimalSeparator == period {
-            return self
+extension String {
+
+    var droppedTrailingZeros: String {
+        var string = self
+
+        while string.last == "0" || string.last == "." {
```

Comment:
Would the "." check cause a problem with locales where the decimal separator is a comma?